src/common/types/vector.cpp

```diff
@@ -403,7 +396,15 @@ void Vector::SetValue(idx_t index, const Value &val) {
 	}
 }
 
-Value Vector::GetValueInternal(const Vector &v_p, idx_t index_p) {
+Value Vector::AttachTypeInfo(const Vector &v, Value &&val) {
+	if (!v.GetType().HasAlias()) {
+		return val;
+	}
+	val.type().CopyAuxInfo(v.GetType());
+	return val;
+}
+
+Value Vector::GetValue(const Vector &v_p, idx_t index_p) {
```

Comment:
Are these changes to `GetValue` necessary? It seems to do the same as the previous function but (in my opinion) in a less readable manner.

Reply:
Yes I agree, the other solution was cleaner. I still need to clean this up from just being functional to also being a less intrusive change